src/framework/dispatchers/RequestDispatcher.kt

```diff
@@ -49,6 +49,8 @@ open class RequestDispatcher: Dispatcher<ApiGatewayRequest, Any> {
 
             if(result.result == SOAResultType.SUCCESS) {
                 return result.data
+            } else if (result.result == SOAResultType.FAILURE && result.message != null) {
+                throw ForbiddenException(result.message as String)
```

Comment:
so is failure always going to be a forbidden exception? i think other stuff can fail later for other reasons

may want to instead throw the error from the controller and have it catch and throw target above (like line 46)